### PR TITLE
Change custom drafts to use id instead of index

### PR DESCRIPTION
**Pull request by RKelln**
Changes custom drafts to use id instead of index.

- Adds draft format helpers to `cubefn.js`.
- 2 other react errors on console for playtest page fixed
- improves format delete route so there aren't weird parameters

### CODE REVIEW

File: src/pages/CubePlaytestPage.js [hunk 1]
```diff
@@ -272,38 +273,26 @@ const CubePlaytestPage = ({ cube, cubeID, canEdit, decks, draftFormats }) => {
...
-          color: 'danger',
-          message: 'Failed to delete format.',
-        });
+        addAlert('danger', `Failed to delete format. ${err.message}`);
       }
     },
     [addAlert, cubeID, formats],
   );
 
-  // Sort formats alphabetically.
```
Comment (phulin) [inline src/pages/CubePlaytestPage.js]: I’d still prefer to do all sorting on the client, even with robust IDs. Layout should be done client side.

Comment (RKelln) [reply src/pages/CubePlaytestPage.js]: Up to you, I agree that sorting should be possible on the client, but generally you should be able to request an order from a server or have it defaulted to a particular deterministic order. If the client changes that order everything should still work (hence the use of IDs and never index).

File: serverjs/cubefn.js [hunk 2]
```diff
@@ -228,6 +228,48 @@ async function getElo(cardnames, round) {
   return result;
 }
 
+function addDraftFormat(cube, { title, multiples, html, packs }) {
```
Comment (joshfinnie) [inline serverjs/cubefn.js]: Could we add tests for these new functions?

Comment (RKelln) [reply serverjs/cubefn.js]: Yup can do.
Can I get some feedback first that this is the style that we want and this is the place where they should live? 
I'm used to CRUD functions being attached to controllers/models, which we don't have, so this is something like that but less elegant.
Also using some newfangled js style here, with destructuring the params and but I think it reads pretty well. Thoughts?

**Comment by phulin** [comment]
It would definitely make sense to me to start attaching these to models.

File: serverjs/cubefn.js (hunk 2 above)
Comment (RKelln) [reply serverjs/cubefn.js]: Tests added.

**Comment by ruler501** [comment]
Now that the new analytics got merged, you may need to check what this change impacts there.

**Comment by RKelln** [comment]
> Now that the new analytics got merged, you may need to check what this change impacts there.

Yup, thanks, will resolve and let you know if I have any questions. :+1:

**Comment by dekkerglen** [comment]
Conflicts

**Comment by RKelln** [comment]
After solving some conflicts I found a few more bugs, but it now makes sense to put the add/update/deleteDraftFormat functions into draftutls.js which will cause conflicts with @dekkerglen's change, so suggest leaving this until after that change is merged in and then I'll fix everything up.

**Comment by RKelln** [comment]
Note to self: analytics also uses custom draft index.

**Comment by dekkerglen** [comment]
Conflicts from draft and deck changes

**Comment by RKelln** [comment]
Yup, this actually has a lot of work left, no need for reviews or anything yet. Will update when this is ready to be looked at again. Thanks.

**Comment by dekkerglen** [comment]
Conflicts from deck and draft updates

**Pull request closed without merging**